indexer-opts/src/lib.rs

```diff
@@ -0,0 +1,358 @@
+                    }
+                }
+            } else {
+                Ok(())
+            }
+        }
+    }
+}
+
+pub fn init_tracing(debug: bool) -> anyhow::Result<tracing_appender::non_blocking::WorkerGuard> {
+    let mut env_filter = EnvFilter::new("indexer_events=info");
+
+    if debug {
+        env_filter = env_filter.add_directive("near_lake_framework=debug".parse()?);
+    }
+
+    if let Ok(rust_log) = std::env::var("RUST_LOG") {
+        if !rust_log.is_empty() {
+            for directive in rust_log.split(',').filter_map(|s| match s.parse() {
+                Ok(directive) => Some(directive),
+                Err(err) => {
+                    tracing::warn!(
+                        target: LOGGING_PREFIX,
+                        "Ignoring directive `{}`: {}",
+                        s,
+                        err
+                    );
+                    None
+                }
+            }) {
+                env_filter = env_filter.add_directive(directive);
+            }
+        }
+    }
+
+    let (non_blocking, guard) = tracing_appender::non_blocking(std::io::stdout());
+
+    let subscriber = tracing_subscriber::fmt::Subscriber::builder()
+        .with_writer(non_blocking)
+        .with_env_filter(env_filter);
+
+    if std::env::var("ENABLE_JSON_LOGS").is_ok() {
+        subscriber.json().init();
+    } else {
+        subscriber.compact().init();
+    }
+
+    Ok(guard)
+}
+
+impl Opts {
+    // returns a Lake Config object where AWS credentials are sourced from .env file first, and then from .aws/credentials if not found.
+    // https://docs.aws.amazon.com/sdk-for-rust/latest/dg/credentials.html
+    pub async fn to_lake_config(
+        &'static self,
+        db_with_meta_data_pool: &sqlx::Pool<sqlx::Postgres>,
+    ) -> anyhow::Result<near_lake_framework::LakeConfig> {
+        let s3_config = aws_sdk_s3::config::Builder::from(&self.lake_aws_sdk_config()).build();
+
+        let config_builder = near_lake_framework::LakeConfigBuilder::default().s3_config(s3_config);
+
+        tracing::info!(target: LOGGING_PREFIX, "CHAIN_ID: {:?}", self.chain_id);
+
+        let start_block_height = match self.start_mode {
+            StartMode::FromLatest => {
+                final_block_height(self.rpc_url.as_ref().unwrap_or_else(|| {
+                    panic!("`rpc-url` must be provided for `--start-mode from-lastest")
+                }))
+                .await
+            }
+            StartMode::FromInterruption => {
+                match last_processed_block_height(&self.indexer_id, db_with_meta_data_pool).await {
+                    Ok(last_processed_block_height) => last_processed_block_height,
+                    Err(err) => {
+                        tracing::warn!(
+                            target: LOGGING_PREFIX,
+                            "Failed to fetch `last_processed_block_height` from meta data. Falling back to provided `start_block_height`\n{:#?}",
```

Comment:
Won't work.

It is actually my way of handling `from-interruption`.

I assume that we start an indexer for the first time with the parameters:

```
--indexer-id my-indexer
--start-block-height 5
--start-mode from-interruption
```

As it is the very first start for `my-indexer` it won't find `latest_processed_block_height` for the first time and will start from the `--start-block-height` provided. 
Next time it is restarted it will find `last_processed_block_height` in the `__meta` and will continue from that value.

If we fail early we won't achieve the desired behavior.

Reply:
You init `latest_processed_block_height` with `start-block-height`, so I still suggest failing early here